pkg/mbset/mbset.go

```diff
@@ -32,24 +32,28 @@ func openMap(name string) (*ebpf.Map, error) {
 }
 
 type state struct {
-	mu     sync.Mutex
-	nextID uint32
+	mu  sync.Mutex
+	ids map[uint32]struct{}
 }
 
 func newState() *state {
-	return &state{}
+	return &state{
+		ids: make(map[uint32]struct{}),
+	}
 }
 
 // AllocID allocates a new ID
 func (s *state) AllocID() (uint32, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	if s.nextID >= MaxIDs {
-		return InvalidID, errors.New("cannot allocate new id")
+
+	for id := range uint32(MaxIDs) {
+		if _, ok := s.ids[id]; !ok {
+			s.ids[id] = struct{}{}
+			return id, nil
+		}
 	}
-	ret := s.nextID
-	s.nextID++
-	return ret, nil
+	return 0, errors.New("cannot allocate new id")
```

Comment:
nit: Previous code returned InvalidID. Not sure if this was changed intentionally or not. No strong feelings either way.

Reply:
we only check for the error, but I think it should return the InvalidID, thanks